legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/platformBinding/typeInfo/typeInfo.pure

```diff
@@ -649,3 +656,89 @@ function  <<access.private>> meta::pure::executionPlan::platformBinding::typeInf
+                                                                                        $leftSide->concatenate($fTypes);,
+                                                            z:Function<Any>[1]|print(if($debug.debug,|$debug.space+'  function: \''+$fe.func.name->toOne()+'\'\n',|''));
+                                                                    let pp = $params->map(v|$v->scanTypesFromAny($processed->concatenate($z),$debug->indent()));
+                                                                    let fTypes= $z->scanTypesFromFunction($processed, $debug->indent());
+                                                                    $pp->concatenate($fTypes);
+                                                                          
+                                                          ]);,
+               i:InstanceValue[1] | print(if($debug.debug,|$debug.space+'Processing Instance'+  + $i->printValueSpecification('') + '\n',|''));
+                                    $i.values->evaluateAndDeactivate()->map( v | $v->scanTypesFromAny($processed, $debug->indent()));
+                                    ,
+               ve:VariableExpression[1] | print(if($debug.debug,|$debug.space+'Processing Variable '+$ve->printValueSpecification('') +'\n',|''));
+                                         $ve.genericType.rawType;,
+               ervs:ExtendedRoutedValueSpecification[1] | print(if($debug.debug,|$debug.space+'Bypassing ExtendedRoutedValueSpecification\n',|''));
+                                                          $ervs.value->scanTypesFromValueSpecification($processed, $debug->indent());,
+               frvs:FunctionRoutedValueSpecification[1] | print(if($debug.debug,|$debug.space+'Bypassing FunctionRoutedValueSpecification\n',|''));
+                                                          $frvs.value->scanTypesFromValueSpecification($processed, $debug->indent());
+             ]
+          );
+
+   print(if($debug.debug,|$debug.space+'Current:'+$res->map(t|$t->toString())->makeString('[',', ',']')+'\n',|''));
+   $res;
+}
+
+function <<access.private>> meta::pure::executionPlan::platformBinding::typeInfo::scanTypesFromFunction(f:Function<Any>[1],processed:Function<Any>[*], debug:DebugContext[1]):Type[*]
+{
+   let fn= if(!$processed->contains($f),|$f,|[]);
+   print(if($debug.debug,|$debug.space+'  - func '+if($fn->isEmpty(), |'not found or excuded', |if($fn.name->isEmpty(), |$fn->toOne()->type().name->toOne(), |$fn.name->toOne()))+'\n',|''));
+   $fn->match([
+      {f0:Function<Any>[0] | []},
+      {f1 :Function<Any>[1] |
+         let paramExtraTypes=  $f1->functionType().parameters->map(p|$p->scanTypesFromAny($processed->concatenate($f1), $debug->indent()));
+         let expressions = $f1->match([l:FunctionDefinition<Any>[1]| $l.expressionSequence, n:NativeFunction<Any>[1]| []]);
```

Comment:
Should we go deep only for user defined functions and not meta:: functions?

Reply:
added logic for same